rclpy/src/rclpy/events_executor/rcl_support.cpp

```diff
@@ -26,7 +26,13 @@ namespace events_executor
 extern "C" void RclEventCallbackTrampoline(const void * user_data, size_t number_of_events)
 {
   const auto cb = reinterpret_cast<const std::function<void(size_t)> *>(user_data);
-  (*cb)(number_of_events);
+  try {
+    (*cb)(number_of_events);
+  } catch (const std::exception & e) {
+    // Catch and print any exception to avoid propagation to c code
+    std::fputs(e.what(), stderr);
```

Comment:
Based on (https://en.cppreference.com/w/c/io/fputs.html):

> The related function [puts](https://en.cppreference.com/w/c/io/puts.html) appends a newline character to the output, while fputs writes the string unmodified.

I think you need a newline character here. Something like:


```suggestion
    std::fputs(e.what(), stderr);
    std::fputs("\n", stderr);
```

Might be easiest, to avoid string manipulation. On the other hand it might cause weird output if this is ever interleaved with other things printing to the console, in which case maybe `fprintf()` is better?
Something like:

```suggestion
    std::fprintf(stderr, "%s\n", e.what());
```

Reply:
Hey @wjwwood,
Thanks for taking the time to look over the code and for pointing this out.
I applied your suggestion.